packages/react/src/components/form/form.stories.tsx

```diff
@@ -35,27 +35,46 @@ interface Values {
 export const Playground: Story<FormProps<Values>> = (props) => (
   <Form {...props}>
     <Field>
-      <Input name="firstName" required>
-        First name <Asterisk aria-label="required" />
-      </Input>
+      <Input
+        name="firstName"
+        label={
+          <>
+            First name <Asterisk aria-label="required" />
+          </>
+        }
+        required
+      />
```

Comment:
This is what I had in mind when we discussed fragments being used in the label.
Personally, I think composition looks much cleaner:

```tsx
<Label>
  First name <Asterisk aria-label="required" />
  <Input name="firstName" required />
</Label>
```

I wonder again if we should even have `label` at all.
The only use case that makes sense to me is:

```tsx
<Input name="firstName" label="First name" required />
```

And even that is a stretch. Doesn't seem much easier or cleaner than:

```tsx
<Label>
  First name
  <Input name="firstName" required />
</Label>
```

Reply:
Similar to previous reply, it would need to still be wrapped in `<span>` (where a "required" indicator is added).

My intentions here was actually to demonstrate yet another way of setting a label. So previously it was with only a string, then with component components, and this - with using "label" prop but as JSX.

Reply:
Right

```tsx
<Label>
  <span>First name <Asterisk aria-label="required" /></span>
  <Input name="firstName" required />
</Label>
```

The `label` prop still looks a bit magic, like, you give it a Fragment but it renders in a `span`.
Wouldn't it make more sense to actually require the wrapping `span` explicitly?

```tsx
<Input
  name="firstName"
  label={
    <span>
      First name <Asterisk aria-label="required" />
    </span>
  }
  required
/>
```

In which case isn't composition still cleaner?

I'm not trying to shoot `[label]` down, I'm really trying to understand where it fits better than `<Label>`.

Reply:
You're right, now when it's just a prop it could just render whatever passed... Totally makes sense.

Main worry - it's a breaking change.

What if we leave this for now, and I'll put a reminder to remove when making a breaking change (that is, when removing "children" prop support).

Reply:
Can't we do something like

```tsx
const Input = ({ children, label }) =>
  <Wrapper>
    {{
      label: children ? <span>{children}</span> : label,
    }}
  </Wrapper>
```

And one final push, is having `[label]` even worth it?
Isn't composition always just better?
Or do you feel like providing several options is what makes it better?

Reply:
You win, refactored that part, so now only deprecating using "children".

Reply:
I didn't want to win 😅  I mean not necessarily.
I'd be happy to agree that flexibility is an advantage in and of itself.

If you're happy without `[label]` then cool.
If you're compromising, then feel free to push back.